### DISCUSSION TITLE
How can I determine fee and token type when sign transaction?

### DISCUSSION THREAD

**Discussion by Sanggon-Kim** [WalletConnect/walletconnect-monorepo#4176] [Developer Support]
I'm developing wallet by WC 2.0.
And I'm getting this data when sign transaction from uniswap.
`{
    "method": "eth_sendTransaction",
    "params": [
        {
            "gas": "0x37852",
            "value": "0x5af3107a4000",
            "from": "0xf04349b4a760f5aed02131e0daa9bb99a1d1d1e5",
            "to": "0x3fc91a3afd70395cd496c647d5a6cc9d4b2b7fad",
            "data": "0x3593564c000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000065b30aff00000000000000000000000000000000000000000000000000000000000000030b080c00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000003000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000c000000000000000000000000000000000000000000000000000000000000001e00000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000005af3107a40000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000005af3107a40000000000000000000000000000000000000000000000000000000023a809154e100000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2000000000000000000000000a3ee21c306a700e682abcdfe9baa6a08f3820419000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000"
        }
    ]
}`

According to WC document, there should be 'gasPrice' value in the data. 
1. How can I get fee with above data? 
2. Is there any way to get token type used in the transaction?

**Comment by glitch-txs** [comment]
could you share a minimal reproducible example?

**Reply by Sanggon-Kim** [reply]
1. Connect your wallet to uniswap [https://app.uniswap.org/](https://app.uniswap.org/) by WalletConnect.
2. Make swap transaction in uniswap page.
3. See the data from uniswap(like above)

**Comment by Avkubs** [comment]
The main thing is to make the visualization user-friendly

**Comment by anasteziyu** [comment]
Hi.   :-)
